open-sphere-base/core/src/main/java/io/opensphere/core/net/NetworkConfigurationManagerImpl.java

```diff
@@ -61,102 +58,130 @@ public void addChangeListener(NetworkConfigurationChangeListener listener)
         myChangeSupport.addListener(listener);
     }
 
-    @Override
-    public String getProxyConfigUrl()
+    /**
+     * Gets the configuration group object, loading it from preferences if
+     * necessary.
+     *
+     * @return the configuration group object, reflecting the persisted state of
+     *         the proxy.
+     */
+    private ProxyConfigurations getConfigurations()
     {
-        return myPrefs.getString(PROXY_CONFIG_URL_PREF_KEY, "");
+        synchronized (PROXY_CONFIG_LOCK)
+        {
+            if (myConfigurations == null)
+            {
+                myConfigurations = myPreferences.getJAXBObject(ProxyConfigurations.class, "configurations", null);
+                // if it's still null, create a new one:
+                if (myConfigurations == null)
+                {
+                    myConfigurations = new ProxyConfigurations();
+                    myConfigurations.setSelectedConfigurationType(ConfigurationType.SYSTEM);
+                    persistConfiguration();
+                }
+            }
+        }
+        return myConfigurations;
     }
 
     @Override
-    public String getProxyExclusions()
+    public void persistConfiguration()
     {
-        return myPrefs.getString(PROXY_EXCLUSION_PATTERNS_PREF_KEY, "");
+        synchronized (PROXY_CONFIG_LOCK)
+        {
+            myPreferences.putJAXBObject("configurations", myConfigurations, false, this);

```

Comment:
Is the saved file structure changing?  If so you will need to add migrators to convert the old config file to the correct format.  Also the config files that are packaged with the installer will need to be updated as well.

Reply:
Yep. We're implementing the migrators as part of another ticket (5659). I'll update the config files in the installers.